spec-tests/runners/kzg.rs

```diff
@@ -0,0 +1,325 @@
+use crate::{
+    test_case::TestCase,
+    test_utils::{load_yaml, Error},
+};
+use ethereum_consensus::deneb::{
+    mainnet::Blob,
+    polynomial_commitments::{
+        blob_to_kzg_commitment, compute_blob_kzg_proof, compute_kzg_proof, verify_blob_kzg_proof,
+        verify_blob_kzg_proof_batch, verify_kzg_proof, FieldElement, KzgCommitment, KzgProof,
+        KzgSettings, ProofAndEvaluation,
+    },
+};
+
+pub fn dispatch(test: &TestCase) -> Result<(), Error> {
+    let kzg_settings = &test.context().kzg_settings;
+
+    match test.meta.handler.0.as_str() {
+        "blob_to_kzg_commitment" => run_blob_to_kzg_commitment_test(test, kzg_settings),
+        "compute_kzg_proof" => run_compute_kzg_proof_test(test, kzg_settings),
+        "verify_kzg_proof" => run_verify_kzg_proof_test(test, kzg_settings),
+        "compute_blob_kzg_proof" => run_compute_blob_kzg_proof_test(test, kzg_settings),
+        "verify_blob_kzg_proof" => run_verify_blob_kzg_proof_test(test, kzg_settings),
+        "verify_blob_kzg_proof_batch" => run_verify_blob_kzg_proof_batch_test(test, kzg_settings),
+        handler => unreachable!("no tests for {handler}"),
+    }
+}
+
+fn run_blob_to_kzg_commitment_test(
+    test: &TestCase,
+    kzg_settings: &KzgSettings,
+) -> Result<(), Error> {
+    let path = &test.data_path;
+    // Load test case ----
+    let path = path.to_string() + "/data.yaml";
+    let test_data: serde_yaml::Value = load_yaml(&path);
+    let input_yaml = test_data.get("input").unwrap();
+    let blob_yaml = input_yaml.get("blob").unwrap();
+    let output_yaml = test_data.get("output").unwrap();
+    let output: Option<KzgCommitment> = serde_yaml::from_value(output_yaml.clone()).unwrap();
+
+    // Check the deserialization of input(s)
+    let blob: Blob = match serde_yaml::from_value(blob_yaml.clone()) {
+        Ok(blob) => blob,
+        Err(_) => {
+            assert!(output.is_none());
+            return Ok(());
+        }
+    };
+
+    let result = blob_to_kzg_commitment(&blob, kzg_settings);
+    if let Some(expected_commitment) = output {
+        // some `output` was present, use inner value to determine if the spec code should succeed
+        // or fail
+        match result {
+            Ok(commitment) => {
+                assert_eq!(commitment, expected_commitment);
+                Ok(())
+            }
+            Err(_) => Ok(()),
```

Comment:
this seems ok but we can be a bit safer by unconditionally `unwrap`ping

```suggestion
        let commitment = result.unwrap();
        assert_eq!(commitment, expected_commitment);
        Ok(())
```

Reply:
and you'll want to update all the other locations